### PR TITLE
Add support for Python 3.8, 3.9

### PR DESCRIPTION
**Pull request by thtrunck**
[sc-132053]
Basic check on 2.7, 3.6, 3.7, 3.8, 3.9.
- create nodepool preset
- create cluster with simplest settings (I only added Service CIDR/DNS IP and Load balancer SKU)
- check monitoring
- run kubectl command (run nginx --image=nginx --restart=Never)
- delete all pods
- test network connectivity
- inspect node pools
- resize cluster
- create containerized exec
- push images
- test containerized exec
- attach cluster

On 3.9 I also tested
- creating the cluster in a different resourcegroup/region
- Using AKS managed identity with and without adding/removing permissions on Vnet/ACR
- Using a different custom identity for the control plane/nodepool
- Using service principal as the cluster identity type

I haven't tested legacy option (but I don't plan to do that).
It would be nice to test GPU but I did my setup in FranceCentral and I'm not able to provision GPU node here.

### CODE REVIEW

File: code-env/python/spec/requirements.txt [hunk 1]
```diff
@@ -1,13 +1,9 @@
...
-azure-mgmt-resource==18.0.0
-azure-mgmt-msi==1.0.0
+azure-identity==1.7.1
+azure-mgmt-authorization==2.0.0
+azure-mgmt-compute==23.1.0
+azure-mgmt-containerservice==16.4.0
+azure-mgmt-resource==20.0.0
+azure-mgmt-msi==6.0.0b1
 ipaddress==1.0.23
-msrest==0.6.21
```
Comment (thtrunck) [inline code-env/python/spec/requirements.txt:11]: Some info about this removal since it was explicitly added in sc-93743
Some exception where moved from msrest into azure-mgmt-core. And we run into that issue because we where pining azure-mgmt-core.

https://github.com/Azure/azure-sdk-for-python/issues/24765#issuecomment-1152709594

**Review by amandineslx — APPROVED** [review]
OK for me globally
Tested on DSS 12.0 (kit from daily build , revision `fd444867f6d5b88f698073369de5327457ed38ba`) with plugin version packaged from this PR

# TL;DR

* globally OK
* ⚠️ We should probably open cards for these two points about the `Resize cluster` macro:
    * default node pool name is not the correct one (`None` instead of `nodepool0`) ; probably not related to this PR
    * node pools can't be deleted with the macro even providing the node pool name ; probably not related to this PR

# Tested

* check where GPU machines are available (type-wise and quota-wise)
```bash
az vm list-skus --all --output table --size Standard_NC --location westus
```
* compare with [Dataiku quotas](https://portal.azure.com/#view/Microsoft_Azure_Capacity/QuotaMenuBlade/~/myQuotas)
* create a new resource group in `westus` region
* perform FM setup and deploy FM
* create an elastic fleet and deploy the design node in DSS 11.4.2
* create a container registry in the same region as DSS(`asouilleuxregistry2.azurecr.io`)
* **Administration > Settings > Containerized execution**
  * create a new **Containerized execution config**
    * **Image registry URL=asouilleuxregistry2.azurecr.io ; Image pre-push hook=Enable push to ACR ; Custom limits=nvidia.com/gpu=1**
* package plugin from PR and upgrade the AKS plugin with this new version
* rebuild containerized execution base images
```bash
# download spark and hadoop zips from https://downloads.dataiku.com/preview/dss/12.0.0-dev21/
# upload them to the machine
rsync -Pav dataiku-dss-*.tar.gz $DSS_TARGET:~/
ssh -i $SSH_KEY $DSS_TARGET
sudo cp dataiku-dss-hadoop-standalone-libs-generic-hadoop3-12.0.0-dev21.tar.gz /opt/dataiku-dss-11.4.3/
sudo cp dataiku-dss-spark-standalone-12.0.0-dev21-3.3.1-generic-hadoop3.tar.gz /opt/dataiku-dss-11.4.3/
# install hadoop and spark support
sudo su dataiku
cd /data/dataiku/dss_data
./bin/dssadmin install-hadoop-integration -standalone generic-hadoop3 -standaloneArchive /opt/dataiku-dss-11.4.3/dataiku-dss-hadoop-standalone-libs-generic-hadoop3-12.0.0-dev21.tar.gz
./bin/dssadmin install-spark-integration -standaloneArchive /opt/dataiku-dss-11.4.3/dataiku-dss-spark-standalone-12.0.0-dev21-3.3.1-generic-hadoop3.tar.gz
# restart DSS instance
./bin/dss restart
# build container exec images
./bin/dssadmin build-base-image --type container-exec --without-r --with-py39 --with-cuda --cuda-version 11.2
# restart DSS instance
./bin/dss restart
```
* `Plugins > ADD PLUGIN`
  * update the plugin with version built from the PR
* `Plugins > Installed > AKS clusters > Code environment > CHANGE`
  * create a new code env **py39** with Python 3.9
* `Administration > Clusters > CREATE AKS CLUSTER`
  * create cluster **asouilleux-cluster** with GPUs :white_check_mark:
    * **Node pools**
      * **Machine type=Standard_NC6s_v3 ; disk size=0 ; Default number of nodes=1 ; Enable nodes autoscaling=ticked ; Min number of nodes=1 ; Max number of nodes=2 ; Availability zones=unticked ; GPU=ticked**
    * **Advanced options**
      * **Service CIDR=10.1.0.0/16 ; DNS IP=10.1.0.10 ; Load balancer SKU=Standard**
* `Administration > Settings > Containerized execution`
  * `Default settings > Default cluster`
    * change value to **asouilleux-cluster**
  * `Resources for Kubernetes containers > Custom limits`
    * add **nvidia.com/gpu=1**
  * `PUSH BASE IMAGES`
* `Administration > Code Envs > NEW PYTHON ENV`
  * create a new code env with Python 3.9
  * `Packages to install`
    * add **tensorflow==2.11.0**
  * `Containerized execution`
    * `Build for > Selected container configurations > azure`
    * `SAVE AND UPDATE`
* create project **sc132053**
  * create a Python Notebook
    * **Code env=py39**
    * **Containerized exec=azure**
  * open Python notebook and check the kernel starts :white_check_mark:
  * execute the following :white_check_mark:
```python
import tensorflow as tf
tf.config.list_physical_devices('GPU')
# [PhysicalDevice(name='/physical_device:GPU:0', device_type='GPU')]
```
* stop the cluster **asouilleux-cluster** :white_check_mark:
* start a cluster **asouilleux-cluster2** with minimal options
  * **Node pools**
    * **Machine type=Standard_B8ms ; disk size=0 ; Default number of nodes=1 ; Enable nodes autoscaling=ticked ; Min number of nodes=1 ; Max number of nodes=2 ; Availability zones=unticked**
  * **Advanced options**
    * **Service CIDR=10.1.0.0/16 ; DNS IP=10.1.0.10 ; Load balancer SKU=Standard**
* `Administration > Settings > Containerized execution`
  * `Resources for Kubernetes containers > Custom limits`
    * remove **nvidia.com/gpu=1**
  * `Default settings > Default cluster`
    * change value to **asouilleux-cluster2**
* `Administration > Clusters > asouilleux-cluster > Actions`
  * `Run kubectl command` :white_check_mark:
  * `Delete finished pods` :white_check_mark:
  * `Delete all pods` :white_check_mark:
  * `Delete finished jobs` :white_check_mark:
  * `Inspect node pools` :white_check_mark:
  * `Resize cluster`
    * :warning: default node pool name is not the correct one (`None` instead of `nodepool0`) ; probably not related to this PR
    * :warning: node pools can't be deleted with the macro even providing the node pool name ; probably not related to this PR
    * resizing the node pool only works :white_check_mark:
  * `Test network connectivity` :white_check_mark:

**Review by amandineslx — APPROVED** [review]
OK for me, see comment above
